### PR TITLE
Separate workload manifests in gather-drop

### PR DESCRIPTION
**Pull request by MilenaHristova**
https://github.com/dotnet/arcade-services/issues/2861

Create a new txt file containing only the manifest packages so that they can be published lastly in the release pipeline
Tested it in the staging pipeline: https://dev.azure.com/dnceng/internal/_build/results?buildId=2254114&view=results 
The new file is published to [pipeline artifacts](https://dev.azure.com/dnceng/internal/_build/results?buildId=2254114&view=artifacts&pathAsName=false&type=publishedArtifacts)

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs [hunk 1]
```diff
@@ -410,6 +410,7 @@ public string GetFileShareLocationForReleaseJson(DownloadedBuild build)
     private const string sympkgsFileName = "sympkgs-all.txt";
     private const string packagesSubDir = "packages";
     private const string symPackagesSubDir = "assets/symbols";
+    private const string manifestCategory = "manifest";
```
Comment (mmitche) [inline src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs:413]: Nit, I think this should be named "core-manifest". At this point we only publish the manifests for packages owned by .net core. If aspnet were to be a future owner though, then we would need a separate list.

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs:413]: I would also move this up near the other category lines: https://github.com/dotnet/arcade-services/pull/2868/files#diff-395e55dce4798af458bafd198854d9fa13c3499a3a8486808cf67407f6788476R333

**Review by mmitche — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs (hunk 1 above)
Comment (MilenaHristova) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs]: Updated

File: src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs [hunk 2]
```diff
@@ -506,9 +510,20 @@ private void CreateReleasePackageLayout(List<DownloadedBuild> downloadedBuilds,
...
+                        && packageFileName.Contains("Workload")
+                        && packageFileName.Contains("Manifest"))
+                    {
+                        StringBuilder categoryStringBuilder = nupkgFileContents.GetOrAddValue(manifestCategory,
+                            () => new StringBuilder());
+                        categoryStringBuilder.AppendLine(relativePackagePath);
+                    }
+                    else
+                    {
+                        StringBuilder categoryStringBuilder = nupkgFileContents.GetOrAddValue(category,
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs:523]: The whole branch seems to differ only in one argument (category vs. manifestCategory) unless I'm missing something. Could you try to refactor the code so there's no duplication of the common parts?

**Review by tkapin — COMMENTED** [review]
Is it possible to write a unit test for this change in the current setup or the testing for this is not set up yet?

File: src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs (hunk 2 above)
Comment (MilenaHristova) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs]: Updated it

**Review by tkapin — APPROVED** [review]

**Comment by MilenaHristova** [comment]
> Is it possible to write a unit test for this change in the current setup or the testing for this is not set up yet?

I couldn't find any tests covering that command

**Comment by mmitche** [comment]
I don't think there is testing for gather-drop.

**Pull request merged**